src/instructlab/provenance/sigstore.py

```diff
@@ -0,0 +1,54 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Standard
+from pathlib import Path
+
+# Third-party
+from sigstore.sign import SigningContext
+from sigstore.oidc import Issuer
+from sigstore.models import Bundle
+from sigstore.verify import Verifier
+from sigstore.verify.policy import Identity
+
+def sign(model_path, bundle_path, staging=False):
+    """Signs a model with Sigstore"""
+    # pylint: disable=C0415
+
+    issuer = Issuer.production()
+    if staging:
+        issuer = Issuer.staging()
+    identity = issuer.identity_token()
+
+    # The artifact to sign
+    artifact = Path(model_path).read_bytes()
+
+    signing_ctx = SigningContext.production()
```

Comment:
I think you need a special case for staging here.

Reply:
Aha, I knew I was missing something. Thanks for the catch!